debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/TransactionalBuffer.java

```diff
@@ -187,39 +178,43 @@ boolean commit(String transactionId, Scn scn, OracleOffsetContext offsetContext,
 
         List<CommitCallback> commitCallbacks = transaction.commitCallbacks;
         LOGGER.trace("COMMIT, {}, smallest SCN: {}, largest SCN {}", debugMessage, smallestScn, largestScn);
-        executor.execute(() -> {
-            try {
-                int counter = commitCallbacks.size();
-                for (CommitCallback callback : commitCallbacks) {
-                    if (!context.isRunning()) {
-                        return;
-                    }
-                    callback.execute(timestamp, smallestScn, scn, --counter);
-                }
+        commit(context, offsetContext, start, commitCallbacks, timestamp, smallestScn, scn, dispatcher);
 
-                lastCommittedScn = Scn.fromLong(scn.longValue());
-            }
-            catch (InterruptedException e) {
-                LogMinerHelper.logError(metrics, "Thread interrupted during running", e);
-                Thread.currentThread().interrupt();
-            }
-            catch (Exception e) {
-                errorHandler.setProducerThrowable(e);
-            }
-            finally {
-                metrics.incrementCommittedTransactions();
-                metrics.setActiveTransactions(transactions.size());
-                metrics.incrementCommittedDmlCounter(commitCallbacks.size());
-                metrics.setCommittedScn(scn.longValue());
-                metrics.setOffsetScn(offsetContext.getScn());
-                metrics.setCommitQueueCapacity(commitQueueCapacity.get());
-                metrics.setLastCommitDuration(Duration.between(start, Instant.now()).toMillis());
-                taskCounter.decrementAndGet();
+        return true;
+    }
+
+    private void commit(ChangeEventSource.ChangeEventSourceContext context, OracleOffsetContext offsetContext, Instant start,
+                        List<CommitCallback> commitCallbacks, Timestamp timestamp, Scn smallestScn, Scn scn, EventDispatcher<?> dispatcher) {
+        try {
+            int counter = commitCallbacks.size();
```

Comment:
Not part of this diff -- but take a look at this method, we do first a get() and then a remove() call on the `transactions` map. I'd expect the latter suffices?

Reply:
Nice catch, method cleaned up and I also took the opportunity to remove `largestScn` from the buffer as was superfluous and no longer necessary.